lib/src/singleton.dart

```diff
@@ -4,5 +4,20 @@
 library grinder.src.singleton;
 
 import '../grinder.dart';
+import 'utils.dart';
 
 final Grinder grinder = new Grinder();
+
+final ZonedValue zonedContext = new ZonedValue(new _NoopContext());
+
+class _NoopContext implements GrinderContext {
+  Grinder get grinder => null;
+
+  GrinderTask get task => null;
+
+  // TODO: implement fail
+  void fail(String message) { }
+
+  // TODO: implement log
```

Comment:
Failing fast could help catch errors, and we can always stop throwing later without it being a breaking change.

Same for fail.